bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/console/InboxConsoleCommandExtension.java

```diff
@@ -100,13 +102,37 @@ public void execute(String[] args, Console console) {
                 case SUBCMD_CLEAR:
                     clearInboxEntries(console, inbox.getAll());
                     break;
+                case SUBCMD_REMOVE:
+                    if (args.length > 1) {
+                        List<DiscoveryResult> results = Stream.of(new DiscoveryResult[] {})
+                                .collect(Collectors.toList());
+                        try {
+                            ThingUID thingUID = new ThingUID(args[1]);
+                            results.addAll(inbox.stream().filter(forThingUID(thingUID)).collect(Collectors.toList()));
+                        } catch (Exception e) {
```

Comment:
This looks nasty - which exception do you need to catch here? If you cannot avoid using try-catch clauses for the program flow, you should at least catch very finegrained exceptions instead - and continue the flow in here.

Reply:
I guess its the `IllegalArgumentException` from the *UID constructor. Another way would be to check if the UID is valid in the given context.

Reply:
Yes, that is to detect a wrong UID.
Look at the other source code in the same file, you will find the same pattern.  I just kept the code homogeneous in the whole file. The only difference here is that I am logging nothing because I was requested to implement a unique command that works for thing id and thing type id without any way to know what kind of ID the user wants to use.

Reply:
The same pattern is actually catching IAE, so please change yours, too. Also, a given UID string can not be both, ThingUID and ThingTypeUID, so please cascade both try/catch blocks. Only when the first try for ThingUID fails with exception you may try to use the argument for a ThingTypeUID.

Reply:
In the approve command, Exception is catched and not IAE. Should I change it ?

Reply:
Yes, please